[DOC] Add documentation for database downloader setting (#210)
logstash-plugins/logstash-filter-geoip: docs/index.asciidoc

The GeoIP filter plugin can manage the database for users running the Logstash default
distribution, or you can manage

> the updates on your own, becoming also responsible for maintaining compliance. The behavior is controlled by the `database` setting.

Reviewer: I'd like to see the note about responsibility moved back to the end of the paragraph.  People tend to read headings, skim through the middle, and read the last sentence. I'm concerned that this important information about responsibility shifting to the user will be overlooked in its current location.

Author: Thank you for the clarification, that's nice to know! I can maybe reword line 50 a bit to include the auto-update and move everything back to the end of the paragraph as you suggested.

```
database updates on your own. The behavior is controlled by the `database` setting and by the auto-update feature.
```

Reviewer: Sounds good.

> When you use the default `database` setting, and the auto-update feature is enabled, Logstash ensures that the plugin is

Reviewer: ```suggestion
When you use the default `database` setting and the auto-update feature is enabled, Logstash ensures that the plugin is
```

Reviewer: Nice clarification! Thanks for adding this detail.

Reviewer: Please remember to either commit the suggestion above (removing first comma) or make the change on your own.

==== Database Auto-update

This plugin bundles Creative Commons (CC) license databases.

> By default, Logstash checks for database updates every day and upgrades from the CC database licence to the EULA version automatically. It downloads the latest version and replaces old databases

Reviewer: Has the behavior changed from the previous version of the docs?  

This wording is confusing because it's mixing a daily update with a one time conversion to the EULA license.

Author: Hum, that's true, this behavior didn't actually change from the previous version. Considering we have now an option to disable the auto-update, I tried to explain here that the license also auto-upgrades from CC to EULA when the download is successfully completed. 

What about changing it to something like: 
```
When the auto-update feature is enabled, Logstash checks for database updates every day. It downloads the latest and can replace the old database
```

In that case, I could also restore the removed NOTE to something like:
```
NOTE: If the auto-update feature is disabled or the database has never been updated successfully, as in air-gapped environments, Logstash can use CC license databases indefinitely.
```

What do you think?

Reviewer: I like where you're heading with this.  If I'm reading your proposed changes correctly, we still need to account for the license conversion.  Something like: 

`If the auto updates feature is enabled, Logstash upgrades from the CC database license to the EULA version on the first download. `

==== Database Auto-update

This plugin bundles Creative Commons (CC) license databases.

> If the auto-update feature is enabled, Logstash checks for database updates every day. It downloads the latest and can replace the old database

Reviewer: ```suggestion
If the auto-update feature is enabled in `logstash.yml`(as it is by default), Logstash checks for database updates every day. It downloads the latest and can replace the old database
```

Reviewer: This is a somewhat unusual situation in that we're talking about related settings that are in two different places (logstash.yml and plugin settings).  It would be good to indicate that the auto-update setting is something else and is enabled by default early on.  The rest of the section makes more sense after we set that initial context.

Otherwise, unless somebody reads the "Disable the auto-update feature" setting info, they won't have this piece of information.

TIP: When possible, allow Logstash to access the internet to download databases so that they are always up-to-date.

**Disable the auto-update feature**

> If you work in air-gapped environment, and want to disable the auto-update feature from Logstash, you can do so by setting the `xpack.geoip.db.auto_update`
> configuration to `false` in `logstash.yml`.

Reviewer: ```suggestion
If you work in air-gapped environment and want to disable the database auto-update feature, set the `xpack.geoip.db.auto_update` value to `false` in `logstash.yml`.
```

> When disabled, a valid `database` location is required, and you are responsible for <<plugins-{type}s-{plugin}-manage_update,manage your own database updates>> and <<plugins-{type}s-{plugin}-database_license,maintaining compliance>>.

Reviewer: ```suggestion
When automatic updates are disabled, a valid `database` location is required.
You are responsible for managing your own database updates and <<plugins-{type}s-{plugin}-database_license,maintaining license compliance>>.
```

Reviewer: Consider removing the link to "maintaining compliance."  It's the very next topic and clicking the link doesn't really take you anywhere.
